cmd/router/main.go

```diff
@@ -281,28 +298,45 @@ func graphHandler(w http.ResponseWriter, req *http.Request) {
 		log.Error(err, "failed to process request")
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(statusCode)
-		w.Write(prepareErrorResponse(err, "Failed to process request"))
+		if _, err := w.Write(prepareErrorResponse(err, "Failed to process request")); err != nil {
+			log.Error(err, "An error has occurred while writing the response")
+		}
 	} else {
 		if json.Valid(response) {
 			w.Header().Set("Content-Type", "application/json")
 		}
 		w.WriteHeader(statusCode)
-		w.Write(response)
+		if _, err := w.Write(response); err != nil {
+			log.Error(err, "an error has occurred while writing the response")
+		}
+	}
+}
 
+func compilePatterns(patterns []string) []*regexp.Regexp {
+	var compiled []*regexp.Regexp
+	for _, p := range patterns {
+		c, err := regexp.Compile(p)
+		if err != nil {
+			log.Error(err, "Failed to compile header pattern", "pattern", p)
```

Comment:
return the error and log outside the function

Reply:
done